src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/UpsertDocument.cs

```diff
@@ -26,7 +26,30 @@ NpgsqlTransaction transaction
 
 public class UpsertDocument(ILogger<UpsertDocument> _logger) : IUpsertDocument
 {
-    private static readonly string _beforeInsertReferences = "BeforeInsertReferences";
+    /// <summary>
+    /// Determine whether invalid referentialIds were descriptors or references, and returns the
+    /// appropriate failure.
+    /// </summary>
+    private static UpsertResult ReportReferenceFailure(
```

Comment:
Looks a lot like `ReportReferenceFailure` for `UpdateDocument()`, but types are different. Extracted out what I could.